public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -574,3 +577,38 @@ function trackIfEmpty(grid: SceneGridLayout) {
     sub.unsubscribe();
   };
 }
+
+/**
+ * Behavior that will capture currently selecteg filters and group by dimensions and save them to local storage,
+ * so that they can be applied when the next dashboard with filters and group by is loaded.
+ * It's working only for dashboards that use filters created automatically for the default data source that supports filtering (newDashboardWithFiltersAndGroupBy feature toggle)
+ */
+function preserveFiltersAndGroupByUrl(scene: DashboardScene) {
+  if (!config.featureToggles.newDashboardWithFiltersAndGroupBy) {
+    return;
+  }
+
+  return () => {
+    const variables = dashboardSceneGraph.getFilterAndGroupByVariables(scene);
+    if (variables.length === 0) {
+      return;
+    }
+    const urlStates: UrlQueryMap = variables.reduce((acc, v) => {
+      // @ts-ignore - accessing protected property
+      const urlState = v['_urlSync'].getUrlState();
```

Comment:
why access the protected property? urlSync is public (and part of the SceenObject interface)

Reply:
My bad!